### PR TITLE
29 parsing architecture

### PR DESCRIPTION
**Pull request by EivindYN**

### CODE REVIEW

File: src/main/java/App.java [hunk 1]
```diff
@@ -91,7 +91,7 @@ public static void main(String[] args) throws Exception {
 
       } catch (Exception e){
         System.out.println("Couldn't delete files..?");
-        System.out.println(e.getStackTrace());
+        e.printStackTrace();
```
Comment (sebastianvitterso) [inline src/main/java/App.java:94]: Forgot about this one, thanks!

File: src/main/java/services/InputParser.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      return ontology;
+    } 
+    SchemaParser parser = null;
+    if (filetype.equals("zip")){
+      parser = new GBFSParser(); //TODO: Add NeTEx if we decide to implement it
+    } else if (filetype.equals("xsd")){
+      parser = new IXSIParser();
+    } else if (filetype.equals("xls") || filetype.equals("xlsx")){
+      parser = new SpreadsheetParser();
+    }
```
Comment (sebastianvitterso) [inline src/main/java/services/InputParser.java:32]: Add an else like:
```
else {
  throw new IllegalArgumentException();
}
```
??

**Review by sebastianvitterso — APPROVED** [review]
Looks good, other than the comment on `InputParser:32`. Feel free to merge if this can be added later!

File: src/main/java/services/InputParser.java (hunk 2 above)
Comment (EivindYN) [reply src/main/java/services/InputParser.java:32]: I can do it, though it should be handled in frontend

Comment (sebastianvitterso) [reply src/main/java/services/InputParser.java:34]: Always do validation [in both front-end and back-end](https://stackoverflow.com/questions/54800075/field-validations-at-frontend-or-backend-which-is-better)!

**Pull request merged**